src/ui/hook/useStepper.ts

```diff
@@ -78,24 +78,25 @@ export const useStepper = (stepsProps: UseStepperInput): UseStepper => {
     }))
   }, [activeStepsId, steps, updateStepStatus])
 
-  const previousStep = useCallback(() => {
-    const activeStep = findStep(steps, activeStepsId.current)
-    if (activeStep.order === 0) return
-
-    const previousStep = steps.find(step => step.order === activeStep.order - 1)
-    const previousStepId = previousStep?.id
-    if (!previousStepId) return
+  const previousStep = useCallback(
+    (isActiveStepValid: boolean) => {
+      const activeStep = findStep(steps, activeStepsId.current)
+      if (activeStep.order === 0) return
 
-    // TODO: add validation check: if active step is valid, update with 'complete' status
-    updateStepStatus(activeStep.id, 'incomplete')
+      const previousStep = steps.find(step => step.order === activeStep.order - 1)
+      const previousStepId = previousStep?.id
+      if (!previousStepId) return
 
-    updateStepStatus(previousStepId, 'incomplete')
+      updateStepStatus(activeStep.id, isActiveStepValid ? 'complete' : 'incomplete')
+      updateStepStatus(previousStepId, 'incomplete')
 
-    setActiveStepsId(prevActiveStepsId => ({
-      current: previousStepId,
-      previous: prevActiveStepsId.current
-    }))
-  }, [activeStepsId, steps, updateStepStatus])
+      setActiveStepsId(prevActiveStepsId => ({
+        current: previousStepId,
+        previous: prevActiveStepsId.current
+      }))
+    },
+    [activeStepsId, steps, updateStepStatus]
+  )
```

Comment:
The changes are applied to line 82 and 90. Other changes relate to code indentation.